application/client.core/src/app/environment/components/sidebar/shell/environment/environment_variables/component.ts

```diff
@@ -0,0 +1,198 @@
+import { Component, ElementRef, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { IEnvironment, INewInformation } from '../component';
+
+import ContextMenuService, { IMenuItem } from '../../../../../services/standalone/service.contextmenu';
+
+@Component({
+    selector: 'app-sidebar-app-shell-environment-variables',
+    templateUrl: './template.html',
+    styleUrls: ['./styles.less']
+})
+
+export class SidebarAppShellEnvironmentVariablesComponent implements OnInit, OnDestroy {
+
+    @Input() public information: INewInformation;
+    @Input() public setEnvironment: (information: INewInformation) => void;
+    @Input() public close: () => {};
+
+    @ViewChild('variableAdd') _ng_variableAdd: ElementRef<HTMLInputElement>;
+
+    public readonly _ng_variable: string = 'variable';
+    public readonly _ng_value: string = 'value';
+    public _ng_selected: IEnvironment;
+    public _ng_adding: boolean = false;
+    public _ng_height: number = 200;
+    public _ng_newVariable: string = '';
+    public _ng_newValue: string = '';
+
+    private _prevValue: string = '';
+    private _prevVariable: string = '';
+    private _editing: IEnvironment | undefined;
+
+    constructor() { }
+
+    public ngOnInit() {
+        this._sortInformation();
+        this._ng_height = Math.round(window.innerHeight * .7);
+    }
+
+    public ngOnDestroy() {
+        if (this._ng_selected !== undefined) {
+            this._ng_selected.editing = {
+                value: false,
+                variable: false
+            };
+            this._ng_selected = undefined;
+        }
```

Comment:
would be `this._ng_selected = undefined` enough? without change of properties and checks. Like
```
public ngOnDestroy() {
   this._ng_selected = undefined;
   this._ng_adding = false;
   this.setEnvironment(this.information);
}
```

Reply:
and do you actually need to drop it as soon as the component would be destroyed in any way? :/

Reply:
The idea behind assigning the editing variables to `false` is to prevent any open inputs for editing/adding.
Since `_ng_selected` is directly connected to the environment, we actually modify the environment. The reason to use `_ng_selected` already has the Object which has the  input open, we save us the looping around all environments to find the one which is still being edited.
But yeah, I guess it makes sense to leave out `this._ng_selected = undefined;` since it's being deleted the moment the component is destroyed